protocol/auditlog.go

```diff
@@ -100,6 +116,7 @@ func (l ConiksAuditLog) Insert(addr string, signKey sign.PublicKey,
 	// create the new directory history
 	h = newDirectoryHistory(addr, signKey, snaps[0])
 
+	// FIXME: remove this check --> caller calls Audit() before this function
```

Comment:
> I don't understand why newDirectoryHistory must be called before Audit, those are independent operations.

It's "Go" things, you are declaring `Audit` as a method of `directoryHistory`, you cannot call Audit without instantiating a variable of `directoryHistory`.

> The current logic is: "Did the audit pass? Yes, so let's insert the snapshots into our log/update our log".

Yes, that's true. I totally agree, but could we say "Do you want to insert into/update our log? Let's check the audit first."

> In fact, at the application level, it would really make more sense to have the auditor simply call Audit() on a new server response and have the insertion/update be abstracted away from the application layer. Then, we can unexport Insert and Update, and pass the snapshots directly instead of the Response

Yes, we are trying to achieve a same thing 🎉 . How can you separate insertion/update if they are abstracted?